### PR TITLE
new: create test functions for scipy-quant

### PR DESCRIPTION
**Pull request by Boniface316**
co-authored-by: Ahmed Darwish <amfa.darwish.97@gmail.com>
co-authored-by: Michał Stęchły <michal.stechly@gmail.com>

## Description

Include description of feature this PR introduces or a bug that it fixes. Include the following information:

Moved scipy-quant from z-quantum to orquestra-opt and created test functions. 

## Please verify that you have completed the following steps

- [X] I have self-reviewed my code.
- [X] I have included test cases validating introduced feature/fix.
- [X] I have updated documentation.

### CODE REVIEW

File: src/orquestra/opt/api/optimizer_test.py [hunk 1]
```diff
@@ -81,6 +82,8 @@ def test_optimizer_succeeds_with_optimizing_sum_of_squares_function(
             sum_x_squared, finite_differences_gradient(sum_x_squared)
         )
 
+        # if self.bounds =
```
Comment (mstechly) [inline src/orquestra/opt/api/optimizer_test.py:85]: Seems that this comment can be deleted.

File: src/orquestra/opt/optimizers/scikit_quant_optimizer.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+class ScikitQuantOptimizer(Optimizer):
+    def __init__(
+        self,
+        method: str,
+        budget: Optional[int] = 10000,
+        bounds: Optional[Union[List[int], List[List[int]], np.ndarray]] = None,
+        recorder: RecorderFactory = _recorder,
+    ) -> None:
+        """
+        Integration with scipy optimizers. Documentation for this module is minimal,
```
Comment (mstechly) [inline src/orquestra/opt/optimizers/scikit_quant_optimizer.py:22]:
```suggestion
        Integration with scikit-quant optimizers. Documentation for this module is minimal,
```

File: src/orquestra/opt/optimizers/scikit_quant_optimizer.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        please refer to https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.minimize.html
```
Comment (mstechly) [inline src/orquestra/opt/optimizers/scikit_quant_optimizer.py:23]: Fix the link please.

File: src/orquestra/opt/optimizers/scikit_quant_optimizer.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+        NOTE:
+            This optimizer doesn't work with one element array as an input. This is due to bug inside the scikit package.
+            Please refer to https://scikit-learn.org/stable/ for sckit package and https://scikit-quant.readthedocs.io/en/latest/ for scikit-quant package.
```
Comment (mstechly) [inline src/orquestra/opt/optimizers/scikit_quant_optimizer.py:27]: This should in the "main docstring", not under the `NOTE` section.

**Review by mstechly — CHANGES_REQUESTED** [review]

File: src/orquestra/opt/api/optimizer_test.py (hunk 1 above)
Comment (Boniface316) [reply src/orquestra/opt/api/optimizer_test.py:85]: Deleted

**Review by mstechly — APPROVED** [review]

File: src/orquestra/opt/optimizers/__init__.py [hunk 5]
```diff
@@ -2,8 +2,16 @@
...
+    from .cma_es_optimizer import CMAESOptimizer
+except ModuleNotFoundError:
+    pass
+try:
+    from .qiskit_optimizer import QiskitOptimizer
+except ModuleNotFoundError:
+    pass
 from .scipy_optimizer import ScipyOptimizer
 from .search_points_optimizer import SearchPointsOptimizer
 from .simple_gradient_descent import SimpleGradientDescent
```
[the remark this answers is not in the document]
Comment (Boniface316) [reply src/orquestra/opt/optimizers/__init__.py:17]: Added

**Comment by Boniface316** [comment]
Implemented the changes from first PR

**Pull request merged**